### DISCUSSION TITLE
tsh hang on macos m1 13.1

### DISCUSSION THREAD

**Discussion by buzzzo** [gravitational/teleport#33219] [Q&A] [answered]
I'm trying to connect from macos using the latest tsh version (no brew version).

When i try to login via tsh it ask me password and mfa but hangs without any log.
The same connections works from linux.

Any ideas ?

Thx

**Comment by webvictim** [comment]
Post the output of `tsh -d login --proxy=teleport.example.com:443` (replace with your own proxy)

**Comment by buzzzo** [comment, marked as answer]
Thx for your quick response

In the logs i've seen that:

023-10-10T22:01:16+02:00 ERRO [CLIENT]    [KEY AGENT] Unable to connect to SSH agent on socket: "/Users/buzzz/.desktop-ssh-agent/agent.sock". error:[

Basically tsh try to connect to the socket of the local ssh-agent, and i've an agent running.
By stopping it tsh was able to connect.

Thx for help

**Reply by webvictim** [reply]
You should also be able to use `tsh login -k no` to suppress adding keys to the running SSH agent on login, if you like.
